ch_pipeline/core/telescope.py

```diff
@@ -554,27 +559,44 @@ def beam(self, feed, freq_id):
             raise ValueError("The requested feed doesn't seem to exist.")
 
         if tools.is_array_x(feed_obj):
-            fname = self.primary_beamx_filename
-
+            pol_ind = list(self._primary_beam.pol[:]).index(b"x")
```

Comment:
@tristpinsm I'm very surprised that you need to use byte strings here as I think that the dataset should be automatically converted to Unicode strings when loaded. 

Also, are these polarisation labels fixed, we tend to use upper case X and Y for polarisations, and I think that it's just confusing if we start to mix.

Reply:
the unicode conversion doesn't apply when you load the container with `ondisk=True`.

Reply:
Ah I see. Good point. Hmmm....